bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/YamlElement.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.model.yaml;
+
+/**
+ * The {@link YamlElement} interface offers an identifier and a check validity method
+ * to any element defined in a YAML configuration file.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+public interface YamlElement {
+
+    /**
+     * Get the identifier of the YAML element
+     *
+     * @return the identifier as a string
+     */
+    String getId();
+
+    /**
+     * Check that the YAML element is valid
+     *
+     * @throws YamlParseException if something is wrong
+     */
+    void checkValidity() throws YamlParseException;
```

Comment:
Why did you choose to throw an exception? I would have expected that a boolean isValid() would be easier to process. You could then just filter them out.

Reply:
There are different checks and this was a way to log a warning containing more detail about the error. If I only return a boolean, the only message I could log "at top" is that the parsing of the YAML file failed.

Or I have to add additional logs for each failing check and in that case I can return a boolean.

Reply:
Replaced by `isValid` returning a boolean